packages/smart/contracts/balancer/BPoolForTesting.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+
+import "./BFactory.sol";
+import "../libraries/ERC20.sol";
+
+contract BPoolForTesting {
+    BFactory public bFactory;
+    uint256 private constant MAX_UINT = 2**256 - 1;
+
+    function createBPoolForTesting(
+        ERC20[] calldata _tokens,
+        uint256[] calldata _initialLiquidity,
+        uint256[] calldata _weights
+    ) external returns (BPool) {
+        require(_tokens.length == _weights.length);
```

Comment:
Also verify that the initial liquidities list is the same length, as well as adding a message explaining why the call failed.